bip-newproposal.mediawiki

```diff
@@ -0,0 +1,607 @@
+3. The security of newer post-quantum schemes is still being evaluated by the cryptographic community
+4. A sudden switch would require extensive coordination across the entire ecosystem
+
+This BIP proposes QES2 as a balanced solution that:
+
+1. Maintains compatibility with existing Bitcoin infrastructure
+2. Adds quantum resistance through a lattice-based signature scheme (Dilithium)
+3. Creates a cryptographic binding between PQC signatures and ECDSA signatures
+4. Allows for a gradual, opt-in transition to quantum-resistant security
+5. Creates a safety net against future quantum computing threats
+
+Specification
+============
+
+Overview
+--------
+
+QES2 combines the Dilithium post-quantum signature scheme with the classical ECDSA signature scheme currently used in Bitcoin. The key innovation is that QES2 doesn't just use both signature schemes in parallel but creates a cryptographic binding between them. First, the Dilithium signature is generated for the transaction digest, then the ECDSA signature signs the Dilithium signature itself. This binding ensures that the ECDSA signature validates the integrity of the quantum-resistant signature.
+
+The QES2 signature process can be represented as:
+
+.. code-block::
+
+   Sign_QES2 = Sign_ECDSA(Sign_PQC(message))
+
+And the verification process:
+
+.. code-block::
+
+   Verify_QES2 = Verify_ECDSA(σ_PQ) ∧ Verify_PQC(message)
+
+This creates a chained trust model where breaking the scheme would require breaking both the quantum-resistant signature and the classical signature.
+
+Mathematical Notation and Parameters
+------------------------------------
+
+Let us define the following notation:
+
+- m: The message to be signed (transaction digest)
+- sk_PQ: Dilithium private key
+- pk_PQ: Dilithium public key
+- sk_EC: ECDSA private key
+- pk_EC: ECDSA public key
+- σ_PQ: Dilithium signature on the message
+- σ_EC: ECDSA signature on the Dilithium signature
+- H: SHA-256 hash function
+- S_PQ: Dilithium signing algorithm
+- V_PQ: Dilithium verification algorithm
+- S_EC: ECDSA signing algorithm
+- V_EC: ECDSA verification algorithm
+
+Dilithium Parameters
+~~~~~~~~~~~~~~~~~~~
+
+The Dilithium instantiation uses the following parameters:
+
+- Security level: NIST security level 3 (equivalent to AES-192)
+- Dimension n: 256
+- Modulus q: 8380417
+- Weight of challenge γ1: 60
+- Weight of secret γ2: (q-1)/16
+- Number of iterations κ: 5
+- Public key size: 1,952 bytes
+- Private key size: 4,000 bytes
+- Signature size: 2,701 bytes
+
+ECDSA Parameters
+~~~~~~~~~~~~~~~
+
+The ECDSA implementation uses the following parameters:
+
+- Curve: secp256k1
+- Field size: 256 bits
+- Public key size: 33 bytes (compressed)
+- Signature size: ~72 bytes (with sighash flag)
+
+Script Types and Address Format
+-------------------------------
+
+This BIP introduces a new script template:
+
+.. code-block::
+
+   <pq_signature_push> <ecdsa_signature_push> <pubkey_push> OP_QES2_CHECKSIG
```

Comment:
Please provide more detail how this new script template would be used in transactions. How is it split between output script, input script, witness section, or a newly introduced transaction section? How would this transaction be serialized? If it is intended to be a soft fork, what mechanism is used to allow unupgraded nodes to accept transactions using this signature scheme?